misc/scripts/dotnet_format.py

```diff
@@ -0,0 +1,19 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+import glob
+import os
+
+# Create dummy generated files.
+for path in [
+    "modules/mono/SdkPackageVersions.props",
+    "modules/mono/glue/GodotSharp/GodotSharp/Generated/GeneratedIncludes.props",
+    "modules/mono/glue/GodotSharp/GodotSharpEditor/Generated/GeneratedIncludes.props",
+]:
+    os.makedirs(os.path.dirname(path), exist_ok=True)
+    with open(path, "w") as f:
+        f.write("<Project />")
+
+# Run dotnet format on every .csproj file.
+for path in glob.glob("**/*.csproj", recursive=True):
+    os.system("dotnet format " + path)
```

Comment:
Is this how the `clang-format` hook works? I'm personally never a big fan of auto-applying changes, I usually have the hook show what's problematic, and prevent the commit 🤔

Reply:
I'm not a big fan of auto-applying changes either, but this should work the same way it worked before:

https://github.com/godotengine/godot/blob/23191b834e4609baacf19855c3acb6a9f607b30f/misc/scripts/dotnet_format.sh#L19-L20

Reply:
That's fair.